pkg/resources/procedure.go

```diff
@@ -53,6 +57,7 @@ var procedureSchema = map[string]*schema.Schema{
 					DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
 						return strings.EqualFold(old, new)
 					},
+					// todo: add validation that this is a valid data type
```

Comment:
in which issue will we handle this? (please add issue number)

Reply:
SNOW-1045437. added

Reply:
Thanks for the issue, I added comment to it, because I remembered (sorry for not bringing it up later), that we have some existing already: `IsDataType()`, `dataTypeValidateFunc`, and `dataTypeDiffSuppressFunc`. Can we use them?

Reply:
You can close it, but please add SNOW-1045437 to the comment content or use one of the mentioned functions

Reply:
i didnt know we had such a function. this should probably be made into a SchemaValdiateDiagFunc though. And instead of having validators in the resources directory, should be moved to a more general validators package

Reply:
updated to use IsDataType() func